src/hooks/usePagination.js

```diff
@@ -1,14 +1,51 @@
+import PropTypes from "prop-types";
+
 export const DOTS = "...";
 
-function usePagination() {
+function usePagination(props) {
```

Comment:
To optimize for performance, the [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) hook can be utilized to avoid expensive calculations on every render.